mlflow/langchain/databricks_dependencies.py

```diff
@@ -166,6 +150,36 @@ def _extract_databricks_dependencies_from_chat_model(chat_model) -> Generator[Re
         yield DatabricksServingEndpoint(endpoint_name=chat_model.endpoint)
 
 
+def _extract_databricks_dependencies_from_tool_nodes(tool_node) -> Generator[Resource, None, None]:
+    from langgraph.prebuilt.tool_node import ToolNode
+
+    if isinstance(tool_node, ToolNode):
+        yield from _extract_databricks_dependencies_from_tools(
+            list(tool_node.tools_by_name.values())
+        )
+
+
+def _extract_databricks_dependencies_from_agent(
+    agent_executor,
+) -> Generator[Resource, None, None]:
+    # Tools are passed into a Langchain as Seq[BaseTool] part of an AgentExecutor
+    # This function looks for an AgentExecutor, extracts all the tools generated from
+    # UC Function Toolkit. From each of these tools it then extracts the Databricks
+    # SQL Warehouse ID and UC Function Names and adds them to resources.
+    from langchain.agents import AgentExecutor
+
+    if isinstance(agent_executor, AgentExecutor):
+        agent = agent_executor.agent
+
+        # Get the Dependencies for the LLM
+        if agent.llm_chain is not None:
+            yield from _extract_databricks_dependencies_from_chat_model(agent.llm_chain.llm)
+            yield from _extract_databricks_dependencies_from_llm(agent.llm_chain.llm)
+
+            tools = agent_executor.tools
+            yield from _extract_databricks_dependencies_from_tools(tools)
+
```

Comment:
Now that the tools is extracted out. Do we need this function? 

Would this resolve this?
```
    yield from _extract_databricks_dependencies_from_chat_model(lc_model)
    yield from _extract_databricks_dependencies_from_retriever(lc_model)
    yield from _extract_databricks_dependencies_from_llm(lc_model)
    if Version(langchain.__version__) >= Version("0.1.0"):
        yield from _extract_databricks_dependencies_from_tool_nodes(lc_model)

    # recursively inspect legacy chain
    for attr_name in _LEGACY_MODEL_ATTR_SET:
        yield from _extract_dependency_list_from_lc_model(getattr(lc_model, attr_name, None))
```

Reply:
Yep updated it to remove it